### ISSUE TITLE
Not compiling at the moment :/

### ISSUE THREAD

**Issue by petemoore** [contester/runlib#5]
@theraphim I'm having problems compiling at the moment - is this related to the recent commits? Thanks!

```
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib/win32 master $ GOOS=windows go install
# github.com/contester/runlib/win32
./win32_windows.go:211: undefined: runtime.KeepAlive
./win32_windows.go:212: undefined: runtime.KeepAlive
./win32_windows.go:213: undefined: runtime.KeepAlive
./win32_windows.go:214: undefined: runtime.KeepAlive
./win32_windows.go:215: undefined: runtime.KeepAlive
./win32_windows.go:216: undefined: runtime.KeepAlive
./win32_windows.go:217: undefined: runtime.KeepAlive
./win32_windows.go:218: undefined: runtime.KeepAlive
./win32_windows.go:219: undefined: runtime.KeepAlive
./win32_windows.go:258: undefined: runtime.KeepAlive
./win32_windows.go:258: too many errors
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib/win32 master $ 
```

**Comment by theraphim** [comment]
You need to update to go 1.7 :)

**Comment by petemoore** [comment]
Awesome, thanks!

**Comment by petemoore** [comment]
@theraphim I spotted this problem today - any ideas what might be up?

```
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib master $ go version && go get -u ./...
go version go1.7.3 darwin/amd64
# github.com/contester/runlib/subprocess
subprocess/subprocess.go:95: undefined: PlatformData
# github.com/contester/runlib/vendor/github.com/Sirupsen/logrus/examples/hook
vendor/github.com/Sirupsen/logrus/examples/hook/hook.go:12: cannot use airbrake.NewHook(123, "xyz", "development") (type *airbrake.airbrakeHook) as type "github.com/contester/runlib/vendor/github.com/Sirupsen/logrus".Hook in argument to log.Hooks.Add:
	*airbrake.airbrakeHook does not implement "github.com/contester/runlib/vendor/github.com/Sirupsen/logrus".Hook (wrong type for Fire method)
		have Fire(*"github.com/Sirupsen/logrus".Entry) error
		want Fire(*"github.com/contester/runlib/vendor/github.com/Sirupsen/logrus".Entry) error
# github.com/contester/runlib/linux
linux/clone_helper.cc:5:10: fatal error: 'sys/capability.h' file not found
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib master $ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib master $ git log -1
commit 72955ff7415fbf3c6568bba3ebb6b90719408e98
Author: Paul Komkoff <i@stingr.net>
Date:   Wed Aug 17 19:23:40 2016 +0100

    :(
pmoore@Petes-iMac:~/go/src/github.com/contester/runlib master $ 
```

From above, I believe I have a clean checkout and am using go 1.7.3...

**Comment by theraphim** [comment]
This was never supposed to work on Mac (darwin).

**Comment by petemoore** [comment]
Thanks @theraphim!

For the linux build, I got things working as follows:

I spotted that there are `.cc` files, so I ran a docker Ubuntu container on my mac, set `CGO_ENABLED` (to be safe) and discovered I needed to `apt-get install libcap-dev` as a dependency, which reduced the build output to just some c++ warnings. The logrus problems were the consequence of unnecessarily building a vendored package which I could avoid by switching the `go get github.com/contester/...` to `go get -u $(go list github.com/contester/... | grep -v /vendor/)`.

Not being designed for darwin makes sense - and the windows problem was also a consequence of `go get github.com/contester/...` building all vendored packages unnecessarily too - so the above fix worked for this too.

So all in all, problems are resolved. Thanks!

I really love this library, by the way - it has been enormously helpful.
